src/Stoa.ts

```diff
@@ -180,8 +180,16 @@ class Stoa extends WebService
                     }
                     else
                     {
-                        preimage_distance = NaN;
-                        result_preimage_hash = Hash.NULL;
+                        if (target_height == new Height(row.enrolled_at))
```

Comment:
It's not a deep comparison, so we'd better modify it like this.
```suggestion
                        if (target_height.value == row.enrolled_at)
```
I got the following result from the code below. 
```
        let h1 = new Height(100n);
        let h2 = new Height(100n);
        console.log(h1 == h2); // false
```